scylla/tests/retries.rs

```diff
@@ -0,0 +1,284 @@
+struct FixedOrderLoadBalancer;
+impl LoadBalancingPolicy for FixedOrderLoadBalancer {
+    fn plan<'a>(
+        &self,
+        _statement: &scylla::load_balancing::Statement,
+        cluster: &'a scylla::transport::ClusterData,
+    ) -> scylla::load_balancing::Plan<'a> {
+        Box::new(
+            cluster
+                .get_nodes_info()
+                .clone()
+                .into_iter()
+                .sorted_by(|node1, node2| Ord::cmp(&node1.address, &node2.address)),
+        )
+    }
+
+    fn name(&self) -> String {
+        "FixedOrderLoadBalancer".to_string()
+    }
+}
+
+async fn test_with_3_node_cluster<F, Fut>(
+    first_proxy_node_addr_last_octet: u16,
+    test: F,
+) -> Result<(), ProxyError>
+where
+    F: FnOnce([String; 3], HashMap<SocketAddr, SocketAddr>, RunningProxy<3>) -> Fut,
+    Fut: Future<Output = RunningProxy<3>>,
+{
+    init_logger();
+    let real1_uri = env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    let proxy1_uri = format!("127.0.0.{}:9042", first_proxy_node_addr_last_octet);
+    let real2_uri = env::var("SCYLLA_URI2").unwrap_or_else(|_| "127.0.0.2:9042".to_string());
+    let proxy2_uri = format!("127.0.0.{}:9042", first_proxy_node_addr_last_octet + 1);
+    let real3_uri = env::var("SCYLLA_URI3").unwrap_or_else(|_| "127.0.0.3:9042".to_string());
+    let proxy3_uri = format!("127.0.0.{}:9042", first_proxy_node_addr_last_octet + 2);
+
+    let real1_addr = SocketAddr::from_str(real1_uri.as_str()).unwrap();
+    let proxy1_addr = SocketAddr::from_str(proxy1_uri.as_str()).unwrap();
+    let real2_addr = SocketAddr::from_str(real2_uri.as_str()).unwrap();
+    let proxy2_addr = SocketAddr::from_str(proxy2_uri.as_str()).unwrap();
+    let real3_addr = SocketAddr::from_str(real3_uri.as_str()).unwrap();
+    let proxy3_addr = SocketAddr::from_str(proxy3_uri.as_str()).unwrap();
+
+    let proxy = Proxy::builder()
+        .with_node(
+            Node::builder()
+                .real_address(real1_addr)
+                .proxy_address(proxy1_addr)
+                .shard_awareness(ShardAwareness::QueryNode)
+                .build(),
+        )
+        .with_node(
+            Node::builder()
+                .real_address(real2_addr)
+                .proxy_address(proxy2_addr)
+                .shard_awareness(ShardAwareness::QueryNode)
+                .build(),
+        )
+        .with_node(
+            Node::builder()
+                .real_address(real3_addr)
+                .proxy_address(proxy3_addr)
+                .shard_awareness(ShardAwareness::QueryNode)
+                .build(),
+        )
+        .build();
+
+    let running_proxy = proxy.run().with_current_subscriber().await.unwrap();
+
+    let mut translation_map = HashMap::new();
+    for (real_addr, proxy_addr) in [
+        (real1_addr, proxy1_addr),
+        (real2_addr, proxy2_addr),
+        (real3_addr, proxy3_addr),
+    ] {
+        translation_map.insert(real_addr, proxy_addr);
```

Comment:
How about adding a method to the proxy which creates this translator/translation map? I think it will be useful in all tests that use proxy and a session.